HopsanCore/src/CoreUtilities/MultiThreadingUtilities.cpp

```diff
@@ -617,6 +617,41 @@ void simWholeSystems(std::vector<ComponentSystem *> systemPtrs, double stopTime)
     }
 }
 
+//! @brief Function for simulating whole systems multi-threaded
+//! @param systemPtrs Vector with pointers to the systems to simulate
+//! @param stopTime Stop time of simulation
+void simWholeSystemInRealtime(double realTimeFactor, volatile bool *pStopSimulation, double *pTime, double timeStep, std::vector<Component*> signalComponentPtrs, std::vector<Component*> cComponentPtrs, std::vector<Component*> qComponentPtrs)
+{
+#if (__cplusplus >= 201103L)
+    auto wallTime = std::chrono::steady_clock::now();
+
+    while(!(*pStopSimulation)) {
+        *pTime += timeStep; //mTime is updated here before the simulation,
+        //mTime is the current time during the simulateOneTimestep
+
+        //Signal components
+        for (const auto sComponent : signalComponentPtrs) {
```

Comment:
not sure if it matters since pointers, but maybe const auto&

Reply:
Of course, it was already used for qComponent. I don't think it matters for pointers, but it should be consistent for s, c and q.